File: Solution-Go/Find_1100.go
func numKLenSubstrNoRepeats(S string, K int) int {
    if len(S) < K {
        return 0
    }
    freq := make(map[rune]int)
    l, r, repeat, res := 0, 0, 0, 0
    for r < len(S) {
        if freq[rune(S[r])] += 1; freq[rune(S[r])] == 2 {
            repeat += 1
        }
        r += 1
        if r - l == K {
            if repeat == 0 {
                res += 1
            }
            if freq[rune(S[l])] -= 1; freq[rune(S[l])] == 1 {
                repeat -= 1
            }
            l += 1
        }
    }
    return res
}

File: Solution-Java/Find_1100.java
class Find_1100 {
    // My first solution: sliding window
    public int numKLenSubstrNoRepeats(String S, int K) {
        if (S.length() < K) return 0;
        int n = S.length();
        int[] freq = new int[26];
        int repeat = 0;
        int left = 0, right = 0;
        int res = 0;
        while (right < n) {
            if (freq[S.charAt(right++) - 'a']++ == 1) repeat++;
            if (right - left == K) {
                if (repeat == 0) res++;
                if (--freq[S.charAt(left++) - 'a'] == 1) repeat--;
            }
        }
        return res;
    }
}